### ISSUE TITLE
Tests which raise exceptions have output with -e but nothing in result file

### ISSUE THREAD

**Issue by dsheets** [mirage/alcotest#32]
```
-- Link.000 Failed --
link.
./_tests/Link.000.output:

[exception] Unix.Unix_error(Unix.ENOENT, "chdir", "link")
Raised by primitive operation at file "test.ml", line 49, characters 4-18
Re-raised at file "test.ml", line 55, characters 10-11
Called from file "test.ml", line 60, characters 17-93
Called from file "test.ml", line 73, characters 31-43
Called from file "test.ml", line 98, characters 2-77
Called from file "lib/alcotest.ml", line 246, characters 8-12
```

but the result file, `_tests/Link.000.output` is 0 bytes.

**Comment by samoht** [comment]
I suppose this is the last version? this redirection mess is annoying...

**Comment by dsheets** [comment]
Yes, the latest version, 0.4.4, at 009cd9d894363e8c5e8a1f024bcf19d9faff3f22. I believe 0397ca05d661bc45e17ede9ab697995c361d6156 was working fine but I haven't explicitly tested it or attempt to bisect.

**Comment by samoht** [comment]
I think that's again a missing flush...

**Comment by dsheets** [comment]
It is actually present in all of 0.4.x.

**Comment by samoht** [comment]
Actually it is present since the beginning of alcotest. The redirection logic is not enable if verbose is set. It's because then we need to use something like `tee` to properly duplicate fd and I'm not sure that's a so good idea in general...

**Comment by samoht** [comment]
See https://github.com/mirage/alcotest/blob/master/lib/alcotest.ml#L303

**Comment by dsheets** [comment]
This occurs even when not using any of `--verbose` or `--show-errors`. The only way to find out why the test failed is to use one of those. I was using `--show-errors` which does not appear to turn on `--verbose` (and they are tracked in the options type separately) but maybe I missed a place where they interact.

**Comment by dsheets** [comment]
I believe I fixed this in #39.
